internal/cp/charset.go

```diff
@@ -2,14 +2,40 @@ package cp
 
 import (
 	"strings"
+	"sync"
 )
 
+var initOnce sync.Once
+
 type charsetMap struct {
 	sb [256]rune    // single byte runes, -1 for a double byte character lead byte
 	db map[int]rune // double byte runes
 }
 
+// lazy initialization of all charset maps
+func initCharsetMaps() {
+	initOnce.Do(func() {
```

Comment:
my guess is most apps only use 1 or 2 collations. Maybe it'd be worthwhile to lazy init them individually instead of all at once.

Reply:
good call. i'll make that change

Reply:
this approach is ok but you could also merge the init into a function that returns the variable.
Something like

```Go
case 30,31,32,33,34:
return getcp437()
```
ideally there'd be no named variable that someone could reference without it first being initialized but I don't know if it's feasible with this model without some unnecessarily exotic dependency inversion.

Reply:
updated from init -> get in [6644c35](https://github.com/microsoft/go-mssqldb/pull/166/commits/6644c35e7a80306176b6d5209c19926efb12301d)